plugins/builds/update.js

```diff
@@ -197,7 +194,8 @@ async function isStageDone({ stage, event }) {
     const stageJobBuilds = await event.getBuilds({ params: { jobId: stageJobIds } });
     let stageIsDone = false;
 
-    if (stageJobBuilds && stageJobBuilds.length !== 0) {
+    // Make sure all builds in stage have run
+    if (stageJobBuilds && stageJobBuilds.length === stageJobIds.length) {
```

Comment:
Number of stage job builds may not always equal to the number of jobs within the stage.
In case of a failure of a stage job, downstream stage jobs will not be executed. In such case, with the changes made teardown may never get executed

Reply:
updated the logic